staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime_test.go

```diff
@@ -144,15 +144,20 @@ func captureStderr(f func()) (string, error) {
 	defer func() { os.Stderr = bak }()
 
 	resultCh := make(chan string)
+	errCh := make(chan error)
```

Comment:
I am not really familiar with the code here, but why is this getting changed?

Reply:
Because the `io.Copy` function returns an error and it was not checked before.

Reply:
sending to two channels in order and receiving from two channels in order in a return is a bit hard to follow.

suggest this instead:

```go
var copyErr error
// copy the output in a separate goroutine so printing can't block indefinitely
go func() {
  ...
  _, copyErr = io.Copy(&buf, r)
  ...
}
...
return <-resultCh, copyErr
```

Reply:
Why should sending and receiving be in order? I think the order doesn't matter and the function returns when both of the channels have received values.

Reply:
> Why should sending and receiving be in order?

return values are evaluated in order, so you are depending on the channel send order inside the goroutine to be the same as the channel receive order in the return statement. That happens to work in your example, but is a slightly weird construction.

https://play.golang.org/p/dbj85uV4bDX is an example of the deadlock that occurs if the send and receive order did not match

Setting the error outside the closure and simply returning that seems more straightforward

Reply:
Thanks for your explanation, I didn't even know it.

Reply:
done. PTAL @liggitt @ahmad-diaa